src/xhr.js

```diff
@@ -89,26 +89,25 @@ class Xhr {
     const sourceOrigin = getSourceOrigin(this.win.location.href);
     const url = parseUrl(input);
     const query = parseQueryString(url.search);
-    if (SOURCE_ORIGIN_PARAM in query) {
-      throw new Error(`Source origin is not allowed in ${input}`);
-    } else {
-      input = addParamToUrl(input, SOURCE_ORIGIN_PARAM, sourceOrigin);
-    }
-    return this.fetch_(input, opt_init).then(response => {
+    assert(!(SOURCE_ORIGIN_PARAM in query), 'Source origin is not allowed in %s',
+        input);
+    input = addParamToUrl(input, SOURCE_ORIGIN_PARAM, sourceOrigin);
+    return this.fetch_(input, opt_init).catch(reason => {
+      assert(false, 'Fetch failed %s: %s', input, reason.message);
```

Comment:
`reason.message` is not guaranteed. Are you ok with this here? Also, we loose the stack in this case, don't we?